src/Services/Wopi/WopiHandler.cs

```diff
@@ -235,25 +236,26 @@ private UserPermissions GetUserPermissions(File file, IUser user)
                 }
                 allowBits = allowBits & ~denyBits;
 
-                //UNDONE: Uncomment this instruction to allow document editing
-                //return new UserPermissions
-                //{
-                //    Write = (allowBits & PermissionType.Save.Mask) > 0,
-                //    RestrictedViewOnly = 0 == (allowBits & PermissionType.Open.Mask) &&
-                //                         0 != (allowBits & (PermissionType.Preview.Mask +
-                //                                            PermissionType.PreviewWithoutWatermark.Mask +
-                //                                            PermissionType.PreviewWithoutRedaction.Mask)),
-                //    Create = !file.Parent?.Security.HasPermission(user, PermissionType.AddNew) ?? false,
-                //};
-                //UNDONE: Delete this instruction to allow document editing
+                if (IsReadOnlyMode)
+                {
+                    return new UserPermissions
+                    {
+                        Write = false,
+                        RestrictedViewOnly = 0 == (allowBits & PermissionType.Open.Mask) &&
+                                             0 != (allowBits & (PermissionType.Preview.Mask +
+                                                                PermissionType.PreviewWithoutWatermark.Mask +
+                                                                PermissionType.PreviewWithoutRedaction.Mask)),
+                        Create = false,
+                    };
+                }
                 return new UserPermissions
                 {
-                    Write = false,
+                    Write = (allowBits & PermissionType.Save.Mask) > 0,
                     RestrictedViewOnly = 0 == (allowBits & PermissionType.Open.Mask) &&
                                          0 != (allowBits & (PermissionType.Preview.Mask +
                                                             PermissionType.PreviewWithoutWatermark.Mask +
                                                             PermissionType.PreviewWithoutRedaction.Mask)),
-                    Create = false,
+                    Create = !file.Parent?.Security.HasPermission(user, PermissionType.AddNew) ?? false,
```

Comment:
Why negate the value?

Reply:
This is wrong. Changed to `false` constant.